usr.sbin/devinfo/devinfo.c

```diff
@@ -179,16 +286,36 @@ int
 print_rman_resource(struct devinfo_res *res, void *arg __unused)
 {
 	struct devinfo_dev	*dev;
-	
-	printf("    ");
-	print_resource(res);
+	struct devinfo_rman *rman;
+	int hexmode;
+	char s[32];
+
 	dev = devinfo_handle_to_device(res->dr_device);
-	if ((dev != NULL) && (dev->dd_name[0] != 0)) {
-		printf(" (%s)", dev->dd_name);
-	} else {
-		printf(" ----");
+	rman = devinfo_handle_to_rman(res->dr_rman);
+	hexmode =  (rman->dm_size > 1000) || (rman->dm_size == 0);
+
+	if (hexmode) {
+		if (res->dr_size > 1)
+			snprintf(s, 32, "0x%lx-0x%lx",
```

Comment:
sizeof(s) is better. but this isn't 32, but really 5 ("0x-0x") + 1 NUL + 2 * (sizeof(long) * 2)  so this can be at most 38, so 32 is a little small.